src/kili/services/export/format/coco/__init__.py

```diff
@@ -157,113 +168,226 @@ def _convert_kili_semantic_to_coco(
-                asset["latestLabel"]["jsonResponse"].items()
-            ):
-                frame_name = f'{asset["externalId"]}_{str(int(frame_id)+1).zfill(leading_zeros)}'
-                coco_image = CocoImage(
-                    id=frame_i + len(assets),  # add offset to avoid duplicate ids
-                    license=0,
-                    file_name=str(DATA_SUBDIR + "/" + f"{frame_name}{frame_ext}"),
-                    height=height,
-                    width=width,
-                    date_captured=None,
-                )
-                coco_images.append(coco_image)
+    for asset in tqdm(assets, desc="Convert to coco format"):
+        nbr_frames = len(asset.get("latestLabel", {}).get("jsonResponse", {}))
+        leading_zeros = len(str(nbr_frames))
+
+        width = height = 0
+        frame_ext = ""
+        # jsonContent with frames
+        if isinstance(asset["jsonContent"], list) and Path(asset["jsonContent"][0]).is_file():
+            width, height = get_image_dimensions(asset["jsonContent"][0])
+            frame_ext = Path(asset["jsonContent"][0]).suffix
+
+        # video with shouldUseNativeVideo set to True (no frames available)
+        elif Path(asset["content"]).is_file():
+            width, height = get_video_dimensions(asset["content"])
+            cut_video(asset["content"], asset, leading_zeros, Path(asset["content"]).parent)
+            frame_ext = ".jpg"
+
+        else:
+            raise FileNotFoundError(f"Could not find frames or video for asset {asset}")
+
+        for frame_i, (frame_id, json_response) in enumerate(
+            asset["latestLabel"]["jsonResponse"].items()
+        ):
+            frame_name = f'{asset["externalId"]}_{str(int(frame_id)+1).zfill(leading_zeros)}'
+            coco_image = CocoImage(
+                id=frame_i + len(assets),  # add offset to avoid duplicate ids
+                license=0,
+                file_name=str(DATA_SUBDIR + "/" + f"{frame_name}{frame_ext}"),
+                height=height,
+                width=width,
+                date_captured=None,
+            )
+            coco_images.append(coco_image)
 
+            if is_single_job:
+                job_name = list(jobs.keys())[0]
                 if job_name not in json_response:
                     coco_img_annotations = []
                     # annotation offset is unchanged
                 else:
                     coco_img_annotations, annotation_offset = _get_coco_image_annotations(
                         json_response[job_name]["annotations"],
-                        cat_kili_id_to_coco_id,
+                        cat_kili_id_to_coco_id[job_name],
                         annotation_offset,
                         coco_image,
                         annotation_modifier,
                     )
                 coco_annotations.extend(coco_img_annotations)
-
+            else:
+                raise NotImplementedError
```

Comment:
why don't we cover the case "is_single_job is False"?

Reply:
I'l add that in this PR then